### ISSUE TITLE
Worldserver.conf - Quests.IgnoreAutoAccept - breaks starting quests

### ISSUE THREAD

**Issue by ceresia** [TrinityCore/TrinityCore#17403]
**Description**:
Starting quests in starting zones (level 1-6) are not allowing characters to accept

**Current behaviour**: Tell us what happens
Upon using the setting "Quests.IgnoreAutoAccept = 1" in worldserver.conf then low level quests in starting zones are not allowed to be manually accepted

**Expected behaviour**: Tell us what should happen instead
Quests should act as any normal quests do of higher level, allow you to click the accept button and roll out!

**Steps to reproduce the problem**:

In my ignorant attempts to recompile last night I reused my worldserver.conf file, figured it would save me time instead of having to input new DB info and such.  After using a fresh one today and figuring out that the following setting was causing the quests to not work:

```
#    Quests.IgnoreAutoAccept
#        Description: Ignore auto accept flag. Clients will have to manually accept all quests.
#        Default:     0 - (Disabled, DB values determine if quest is marked auto accept or not.)
#                     1 - (Enabled, clients will not be told to automatically accept any quest.)

Quests.IgnoreAutoAccept = 0
```

If I set Quests.IgnoreAutoAccept = 1  then no starting quests will work at all.  

So instead of the bug being that the quests dont work it is that the "Quests.IgnoreAutoAccept = 1" breaks being able to accept starting quests.

**Branch(es)**: 3.3.5

**TC hash/commit**:  
TrinityCore rev. 90ff3ce 2016-06-16 17:53:14 +0200 (3.3.5 branch) (Win32, Release)

**TDB version**:  
335.61

**Operating system**:  
Host: Server 2008R2 
Client: Windows 10 & Windows 7 both tested.

**Comment by tkrokli** [comment]
TrinityCore rev. 736843fbb544 2016-06-17 13:28:26 +0200 (3.3.5 branch) (Win64, Release) (Static)
TDB 335.61 plus updates up to and including `2016_06_17_00_world.sql`

Confirmed, just tested this on 4 new Alliance and Horde characters (dwarf, night elf, tauren, troll). To be more specific about **Current behaviour**, clicking the [Accept] button does nothing except for closing the Gossip window, as if the source code does not apply the expected Quest.Accept action, but only uses `PlayerTalkClass->SendCloseGossip()` to close the gossip window.

Some more information related to this issue; I believe this is what it has been like since the Worldserver.conf option got introduced in the PR **Core/Miscellaneous Feature Quest Auto Accept/Ignore #5150** from February 9, 2012 by @Kline- . If you read the last comment in that PR, the author says that this behaviour is intended (don't know if Kline- understood Expecto's feedback details, though).

Also this issue, only for completeness: **Accept button in quest menu not works #11500**

From what I can remember 2 years back, if this option in worldserver.conf was enabled, the [Accept] button usually did not react at all, it did not even close the gossip window and I had to pres ESC or click the [X] in the top right corner to close it. I did not think much of this option before, so I wonder if it has ever worked to accept quests **manually** when the quests themselves got the AutoComplete status.

Source code: https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Quests/QuestDef.cpp#L233-L236
```cpp
bool Quest::IsAutoAccept() const
{
    return !sWorld->getBoolConfig(CONFIG_QUEST_IGNORE_AUTO_ACCEPT) && HasFlag(QUEST_FLAGS_AUTO_ACCEPT);
}
```
I am hoping to find out where this is handled in the source and see if we can get the [Accept] button to work when `Quests.IgnoreAutoAccept = 1` is set. So until further changes, this is what it has been like since February 2012 & commit https://github.com/TrinityCore/TrinityCore/commit/d2e018dbbc7b207c0b7370df4510e17f803e8bef .

**Comment by adeutscher** [comment]
When investigating this I tested with the Orc/Troll starting quest (ID 4641) that had the auto-complete flag and a quest from Sen'Jin Village that did not.

I noticed in my log that the auto-complete quest does not trigger a log message with "CMSG_QUESTGIVER_ACCEPT_QUEST" (from `WorldSession::HandleQuestgiverAcceptQuestOpcode`, beginning at https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Handlers/QuestHandler.cpp#L101 ). This suggests to me that the client is saying to itself "I was told that this quest will be added for me, so I am not going to bother with sending a message to accept it (never mind what button that meatbag pressed)".

My initial idea for fixing this looks like this (to be placed in place of https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Entities/Creature/GossipDef.cpp#L412):

```cpp
    // Rough proof of concept scribbles for fixing manually accepting quests while auto-acceptance is ignored.
    if(sWorld->getBoolConfig(CONFIG_QUEST_IGNORE_AUTO_ACCEPT))
        data << uint32(quest->GetFlags() & ~QUEST_FLAGS_AUTO_ACCEPT); // 3.3.3 questFlags
    else
        data << uint32(quest->GetFlags());                  // 3.3.3 questFlags
```

For fixing this in a cleaner manner, perhaps the quest's flags could be stripped down in the post-processing section of `ObjectMgr::LoadQuests` ( https://github.com/TrinityCore/TrinityCore/blob/3.3.5/src/server/game/Globals/ObjectMgr.cpp#L4170-L4714 ) so that extra processing is only done when the quests are initially loaded.
